### PR TITLE
Inline util.format and replace usage of global

### LINKED ISSUE #1999: BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
### Describe the problem
 After upgrading angular to version 12, which uses webpack 5 I got the following error,

```

./node_modules/auth0-lock/lib/i18n.js:24:12-27 - Error: Module not found: Error: Can't resolve 'util' in 'C:\Users\mikim\WebstormProjects\reynapro-plugin\
node_modules\auth0-lock\lib'


BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
        - install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "util": false }
```

### What was the expected behavior?

 The angular application should compile, but it does not because of auth0-lock dependencies...

This is brand new project with angular 12, just run `npm run start` command, this will appear:

image

> Where possible, please include the sample app:
>
> - https://github.com/miki995/test-auth0-lock-angular12

- Step 1: Open this app, run `npm install`
- Step 2: Run `npm run start`
- ...

### Environment

- **Version of this library used:** Latest, 11.30.0
- **Which framework are you using, if applicable:** Angular 12
- **Other modules/plugins/libraries that might be involved:** auth0-js
- **Any other relevant information you think would be useful:**

### PR DESCRIPTION
**Pull request by stevehobbsdev**
### Changes

This PR inlines the usage of `util.format` and removes the need to pull in the native `util` module. This fixes environments like Webpack 5 that no longer automatically polyfills such native modules.

It also replaces the usage of `global` with `window`, the again the defaults around providing this in Webpack 5 have changed. However, this library was incorrectly using `global` where it should have just used `window`.

### References

Fixes #1999

### Testing

* [X] This change adds unit test coverage
* [ ] This change adds integration test coverage
* [X] This change has been tested on the latest version of the platform/language

### Checklist

* [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
* [X] All code quality tools/guidelines have been run/followed
* [X] All relevant assets have been compiled

### CODE REVIEW

File: src/__tests__/connection/enterprise/__snapshots__/hrd_screen.test.js.snap [hunk 1]
```diff
@@ -5,7 +5,7 @@ exports[`HRDScreen Component renders correctly when there is an enterprise domai
   data-__type="hrd_pane"
   data-header={
     <p>
-      Please enter your corporate credentials at [ 'domain.com' ].
+      Please enter your corporate credentials at domain.com.
```
Comment (adamjmcgrath) [inline src/__tests__/connection/enterprise/__snapshots__/hrd_screen.test.js.snap:8]: I'm assuming it's ok that these snapshots have changed? The new output looks better...

**Review by adamjmcgrath — APPROVED** [review]

File: src/__tests__/connection/enterprise/__snapshots__/hrd_screen.test.js.snap (hunk 1 above)
Comment (stevehobbsdev) [reply src/__tests__/connection/enterprise/__snapshots__/hrd_screen.test.js.snap:8]: These appear to change depending on the version of node that generates the snapshots. I think with Node 10 you get the former version, but newer versions omit the wrapping `[' ']`. The output on the actual screen is the same though. These just snuck in from a previous check-in.

**Comment by GlebVST** [comment]
Hey @stevehobbsdev so apparently this change from `global` to `window` could break some apps using auth0 lock (as it did our) in server-side-rendering environment (where you have no window basically), you guys sure this was good for a minor version change??

**Comment by stevehobbsdev** [comment]
@GlebVST Sorry to hear that we may have inadvertently broken you; we believed it to be a fairly innocuous change given the use cases we support. Which SSR environment/framework are you using?

SSR isn't an officially supported use case for Lock; normally I would suggest configuring your app in such a way that it ignores Lock for the SSR build, if that's possible.

**Comment by GlebVST** [comment]
@stevehobbsdev err it's a bit of surprise to know SSR is not supported, after like 4 years we're using it ;) Yeah it's a bit dated React app where Lock is just used on a login component that gets rendered by node/express when people land on /login path - not quite sure why such usage would be out of support? 
Right, rolled back to 11.30.4 for now...

**Pull request merged**